docs/source/getting-started/what-is-jupyterhub.md

```diff
@@ -0,0 +1,465 @@
+files by other extensions to the single-user notebook servers or
+configuring the spawners to access shared storage spaces.  In order to
+use something such as hubshare, the difficulty is not modifying
+JupyterHub: it is modifying the notebook servers to speak to some
+service, and making that service.
+
+The configuration option `c.JupyterHub.services` is used to start
+services from the hub.
+
+When a service is started from JupyterHub automatically, its logs are
+included in the JupyterHub logs.
+
+
+
+## Single-user notebook server
+
+The **single-user notebook server** is the same thing you get by
+running `jupyter notebook` or `jupyter lab` from the command line -
+the actual Jupyter user interface for a single person.
+
+The role of the spawner is to start this server - basically, running
+the command `jupyter notebook`.  Actually it doesn't run that, it runs
+`jupyterhub-singleuser` which first communicates with the hub to say
+"I'm alive" before running a completely normal Jupyter server.  The
+single-user server can be JupyterLab or classic notebooks.  By this
+point, the hub is almost completely out of the picture (the web
+traffic is going through proxy unchanged).  Also by this time, the
+spawner has already decided the environment which this single-user
+server will have and the single-user server has to deal with that.
+
+The spawner starts the server using `jupyterhub-singleuser` with some
+environment variables like `JUPYTERHUB_API_TOKEN` and
+`JUPYTERHUB_BASE_URL` which tell the single-user server how to connect
+back to the hub in order to say that it's ready.
+
+The single-user server options are **JupyterLab** and **classic
+Jupyter Notebook**.  Really, there isn't that much difference between
+them, they run through the same backend server process and the web
+frontend is an option when it is starting.  The spawner can choose the
+command line when it starts the single-user server.  Extensions are a
+property of the single-user server (in two parts: there can be a part
+that runs in the Python server process, and parts that run in
+javascript in lab or notebook).
+
+If one wants to install software for users, it is not a matter of
+"installing it for JupyerHub" - it's a matter of installing it for the
+single-user server, which might be the same environment as the hub,
+but not necessarily.  (Actually, see below - it's a matter of the
+kernels!)
+
+After the single-user notebook server is started, any errors are only
+an issue of the single-user notebook server.  Sometimes, it seems like
+the spawner is failing, but really the spawner is working but the
+single-user notebook server dies right away (in this case, you need to
+find the problem with the single-user server and adjust the spawner to
+start it correctly or fix the environment).  This can happen, for
+example, if the spawner doesn't set an environment variable or doesn't
+provide storage.
+
+The single-user server's logs are handled by the spawner, so if you
```

Comment:
Maybe I'm confused by what you mean by "handled by the spawner" but I didn't think this was correct.  The notebook server itself does its own logging, however it's supposed to do that.  So for instance a notebook setup with BatchSpawner just logs to stdout and appears in the batch job output.  Maybe you meant something else here?